UQL/Interpreter.php

```diff
@@ -50,18 +51,18 @@ class Interpreter
      * Constructor needs the columns descriptor to figure out appropriate filtering methods
      * and translate identifiers.
      *
-     * @param UqlExtensionContainer  $extensionContainer
+     * @param UqlFunctionCaller      $uqlFunctionCaller
      * @param DataSourceInterface    $dataSource
      * @param FilterConditionFactory $filterConditionFactory
      * @param bool                   $caseSensitive
      */
     public function __construct(
-        UqlExtensionContainer $extensionContainer,
+        UqlFunctionCaller $uqlFunctionCaller,
```

Comment:
maybe breaks bc?
this could also limit future functionality: at the moment we only have functions as extensions. what happens if we need to access other extensions in the enterpreter in the future?